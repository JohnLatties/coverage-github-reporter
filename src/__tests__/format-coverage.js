/* eslint-env jest */

const { format } = require('../format-coverage')

describe('format', () => {
  it('formats report', () => {
    expect(format(require('./basic-report.json'))).toMatchSnapshot()
  })

  it('formats report with links', () => {
    expect(format(
      require('./basic-report.json'),
      undefined,
      'http://example.com/artifacts'
    )).toMatchSnapshot()
  })

  it('formats delta report', () => {
    expect(format(
      require('./basic-report2.json'),
      require('./basic-report.json')
    )).toMatchSnapshot()
  })

  it('formats delta report with links', () => {
    expect(format(
      require('./basic-report2.json'),
      require('./basic-report.json'),
      'http://example.com/artifacts'
    )).toMatchSnapshot()
  })
})
