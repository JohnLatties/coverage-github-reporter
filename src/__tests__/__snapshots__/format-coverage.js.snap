// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`format formats delta report 1`] = `
"Project Coverage    18.57% 💔   (no change)
<pre>
src/                    10.00% 💔    -8.57% 😭
  bot.js               100.00% ✅  +100.00% 🎉
  cli.js                60.00% 💛   +60.00% 😍
  format-coverage.js    35.00% 💔    -1.13% 😥
  github-comment.js     10.00% 💔   +10.00% 🙂
  parse-coverage.js      0.00% ❌   -13.64% 😱
  new-file.js            0.00% ❌ 
</pre>
<details>
<summary>Folder Coverage</summary>
<pre>
src/    10.00% 💔 
</pre>
</details>"
`;

exports[`format formats delta report 2 1`] = `
"Project Coverage    18.57% 💔   (no change)
<pre>
src/        20.00% 💔   +10.00% 🙂
  bot.js    80.00% 💚   -20.00% 😡
</pre>
<details>
<summary>Folder Coverage</summary>
<pre>
src/    20.00% 💔 
</pre>
</details>"
`;

exports[`format formats delta report with links 1`] = `
"<a href=\\"http://example.com/artifacts/index.html\\">Project Coverage</a>    18.57% 💔   (no change)
<pre>
<a href=\\"http://example.com/artifacts/src/index.html\\">src/                </a>    10.00% 💔    -8.57% 😭
<a href=\\"http://example.com/artifacts/src/bot.js.html\\">  bot.js            </a>   100.00% ✅  +100.00% 🎉
<a href=\\"http://example.com/artifacts/src/cli.js.html\\">  cli.js            </a>    60.00% 💛   +60.00% 😍
<a href=\\"http://example.com/artifacts/src/format-coverage.js.html\\">  format-coverage.js</a>    35.00% 💔    -1.13% 😥
<a href=\\"http://example.com/artifacts/src/github-comment.js.html\\">  github-comment.js </a>    10.00% 💔   +10.00% 🙂
<a href=\\"http://example.com/artifacts/src/parse-coverage.js.html\\">  parse-coverage.js </a>     0.00% ❌   -13.64% 😱
<a href=\\"http://example.com/artifacts/src/new-file.js.html\\">  new-file.js       </a>     0.00% ❌ 
</pre>
<details>
<summary>Folder Coverage</summary>
<pre>
<a href=\\"http://example.com/artifacts/src/index.html\\">src/</a>    10.00% 💔 
</pre>
</details>"
`;

exports[`format formats delta report with no changes 1`] = `
"Project Coverage    18.57% 💔   (no change)
<details>
<summary>Folder Coverage</summary>
<pre>
src/    10.00% 💔 
</pre>
</details>"
`;

exports[`format formats report 1`] = `
"Project Coverage    18.57% 💔 
<pre>
src/    18.57% 💔 
</pre>
<details>
<summary>Folder Coverage</summary>
<pre>
src/    18.57% 💔 
</pre>
</details>"
`;

exports[`format formats report with links 1`] = `
"<a href=\\"http://example.com/artifacts/index.html\\">Project Coverage</a>    18.57% 💔 
<pre>
<a href=\\"http://example.com/artifacts/src/index.html\\">src/</a>    18.57% 💔 
</pre>
<details>
<summary>Folder Coverage</summary>
<pre>
<a href=\\"http://example.com/artifacts/src/index.html\\">src/</a>    18.57% 💔 
</pre>
</details>"
`;
